### DISCUSSION TITLE
How to transfer vector type material parameters to VectorPostprocessor from Materials

### DISCUSSION THREAD

**Discussion by PengWei97** [idaholab/moose#25897] [Q&A General] [answered]
Dear MOOSE Community,

I hope this message finds you well, and I'm currently continuing to complete goal #25687. As part of this task, I am attempting to access a material property of type `MaterialProperty<std::vector<Real>> _slip_resistance`  within the [CrystalPlasticityStressUpdateBase](https://github.com/idaholab/moose/blob/bee12eebdc78767310c651a3b4ceed05f9115ad2/modules/tensor_mechanics/src/materials/crystal_plasticity/CrystalPlasticityStressUpdateBase.C#L240) material class, through the use of a VectorPostprocessor and named [FeatureMatePropVectorPostprocessorV2](https://github.com/PengWei97/qinglong/commit/388a7a8295ab956e411b75102abdd6dff3f601c4).

FeatureMatePropVectorPostprocessorV2.h
```c++
  const MaterialProperty<Real> & _test_real;
  const MaterialProperty<std::vector<Real>> & _slip_resistance;
```

FeatureMatePropVectorPostprocessorV2.C
```c++
FeatureMatePropVectorPostprocessorV2::FeatureMatePropVectorPostprocessorV2(
    const InputParameters & parameters)
  : FeatureVolumeVectorPostprocessorCopy(parameters),
    _mat_prop_qp(getMaterialProperty<Real>("mat_prop")),
    _slip_resistance(getMaterialProperty<std::vector<Real>>("slip_resistance")),
    _test_real(getMaterialProperty<Real>("test_real")),
    _mat_prop_aves(declareVector("mat_prop_aves")),
    _grain_id_to_slip_resistances_map()
{
  std::cout << "the done of FeatureMatePropVectorPostprocessorV2" << std::endl;
}

void
FeatureMatePropVectorPostprocessorV2::initialize()
{
  std::cout << "FeatureMatePropVectorPostprocessorV2::initialize()" << std::endl;
  std::cout << "_test_real: " << _test_real[0] << std::endl;
  std::cout << "_slip_resistance["<< 0 << "].size() " << _slip_resistance[0].size() << std::endl;
}
```
Output results:

![image](https://github.com/idaholab/moose/assets/56183775/fc21aef0-c6e9-4ad0-a66c-24cdb2c77d7a)

However, despite my efforts, I have encountered difficulties in accessing the `_slip_resistance` property. Strangely, I have found that I can successfully retrieve a material property of type `MaterialProperty<Real>` (e.g., `_test_real`), but I am facing challenges when attempting to access `std::vector<Real>`-type material properties, such as `_slip_resistance`.

I would greatly appreciate your guidance and insights into this matter. I am looking for a solution or suggestions on how to access and work with material properties of the `std::vector<Real>` type through a VectorPostprocessor. Understanding how to access and manipulate these properties is crucial for the successful completion of my task.

If you require any additional information or code snippets to better understand the issue, please do not hesitate to ask. Your assistance in resolving this matter will be immensely valuable.

Thank you in advance for your time and support. I look forward to your response and guidance.

Best regards,

Wei

**Comment by GiudGiud** [comment, marked as answer]
Hello

What kind of challenges are you facing?
the interfaces should be roughly the same, except maybe that Real is replaced by `std::vector<Real>`

**Reply by PengWei97** [reply]
Hi @GiudGiud, thank you very much for your attention. This issue has been resolved. And in a while, I will elaborate on the solution. At the same time, there is a new issue #25906, hoping to get your guidance.
